diff -crNB openidm-ui-ria-admin/css/apps/dashboard.less openidm-ui-ria-admin-apache/css/apps/dashboard.less
*** openidm-ui-ria-admin/css/apps/dashboard.less	1970-01-01 00:00:00.000000000 +0000
--- openidm-ui-ria-admin-apache/css/apps/dashboard.less	2012-12-01 19:33:33.362896676 +0000
***************
*** 0 ****
--- 1,19 ----
+ ul.appsDashboard {
+         list-style-type: none;
+         li {
+                 display: inline-block;
+                 border: solid thin #DBDBDB;
+                 padding: 5px;
+                 margin: 0 10px;
+ 		text-align: center;
+ 
+ 		img {
+ 			display: block;
+ 			margin: 0 auto;
+ 			padding-bottom: 3px;
+ 			max-width: 100px;
+ 			max-height: 100px;
+ 		}
+         }
+ }
+ 
diff -crNB openidm-ui-ria-admin/css/styles.less openidm-ui-ria-admin-apache/css/styles.less
*** openidm-ui-ria-admin/css/styles.less	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/css/styles.less	2012-12-01 18:50:03.744329662 +0000
***************
*** 34,37 ****
  /* specific css used in specific modules */
  @import "user/forms.less";
  @import "admin/task-table.less";
! @import "admin/notifications.less";
\ No newline at end of file
--- 34,38 ----
  /* specific css used in specific modules */
  @import "user/forms.less";
  @import "admin/task-table.less";
! @import "admin/notifications.less";
! @import "apps/dashboard.less";
diff -crNB openidm-ui-ria-admin/css/user/layout.less openidm-ui-ria-admin-apache/css/user/layout.less
*** openidm-ui-ria-admin/css/user/layout.less	2012-11-28 05:09:50.000000000 +0000
--- openidm-ui-ria-admin-apache/css/user/layout.less	2012-12-01 18:45:44.153178885 +0000
***************
*** 266,269 ****
         display: block;
  	   font-size: 15px;
      }
! }
\ No newline at end of file
--- 266,271 ----
         display: block;
  	   font-size: 15px;
      }
! }
! 
! 
diff -crNB openidm-ui-ria-admin/locales/en-US/translation.json openidm-ui-ria-admin-apache/locales/en-US/translation.json
*** openidm-ui-ria-admin/locales/en-US/translation.json	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/locales/en-US/translation.json	2012-11-30 14:14:59.289093355 +0000
***************
*** 70,76 ****
                  "noAccountQuestion" : "Don't have an account?",
                  "registerAccount" : "Register your account",
                  "problemLoggingQuestion" : "Having trouble logging in?",
!                 "resetPassword" : "Reset my password",
                  "siteIdentificationHeader": "Site Identification",
                  "siteIdentificationExplanation": "Please enter your login name.<br />Your site image and pass phrase will be displayed here."
              },
--- 70,76 ----
                  "noAccountQuestion" : "Don't have an account?",
                  "registerAccount" : "Register your account",
                  "problemLoggingQuestion" : "Having trouble logging in?",
!                 "resetPassword" : "Reset your password",
                  "siteIdentificationHeader": "Site Identification",
                  "siteIdentificationExplanation": "Please enter your login name.<br />Your site image and pass phrase will be displayed here."
              },
***************
*** 283,288 ****
--- 283,293 ----
              }
           }
      },
+     "openam" : {
+         "apps" : {
+             "header" : "My Applications"
+         }
+     },
      "openidm" : {
          "ui" : {
              "common" : {
diff -crNB openidm-ui-ria-admin/main.js openidm-ui-ria-admin-apache/main.js
*** openidm-ui-ria-admin/main.js	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/main.js	2012-11-30 03:32:59.126405676 +0000
***************
*** 120,126 ****
      "config/main",
      "org/forgerock/commons/ui/user/main",
      "org/forgerock/commons/ui/common/main",
!     "org/forgerock/openidm/ui/admin/main" 
  ], function(a, b, c, d, e, f, g, h, i, j, k, l, m, n, i18n, constants, eventManager) { 
      eventManager.sendEvent(constants.EVENT_DEPENDECIES_LOADED);
  });
--- 120,127 ----
      "config/main",
      "org/forgerock/commons/ui/user/main",
      "org/forgerock/commons/ui/common/main",
!     "org/forgerock/openidm/ui/admin/main",
!     "org/forgerock/openam/ui/apps/main" 
  ], function(a, b, c, d, e, f, g, h, i, j, k, l, m, n, i18n, constants, eventManager) { 
      eventManager.sendEvent(constants.EVENT_DEPENDECIES_LOADED);
  });
diff -crNB openidm-ui-ria-admin/org/forgerock/commons/ui/common/main/ErrorsHandler.js openidm-ui-ria-admin-apache/org/forgerock/commons/ui/common/main/ErrorsHandler.js
*** openidm-ui-ria-admin/org/forgerock/commons/ui/common/main/ErrorsHandler.js	2012-11-28 05:09:34.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/commons/ui/common/main/ErrorsHandler.js	2012-11-30 21:38:45.678354846 +0000
***************
*** 58,65 ****
          }
          
          if(handler) {
!             // conditional check needed here until calls to authentication?_action=reauthenticate no longer produce 403 status
!             if (!error.hasOwnProperty("responseObj") || !(error.responseObj.error === 403 && error.responseObj.message === "Reauthentication failed")) {
                  if(handler.event) {
                      eventManager.sendEvent(handler.event, {handler: handler, error: error});
                  }
--- 58,66 ----
          }
          
          if(handler) {
!             // conditional check needed here until calls to authentication?_action=reauthenticate and OpenAM authentication no longer produce 403 status
!             if (!error.hasOwnProperty("responseObj") || !(error.responseObj.error === 403 && 
!                 (error.responseObj.message === "Reauthentication failed" || error.responseObj.message === "SSO Token cannot be retrieved."))) {
                  if(handler.event) {
                      eventManager.sendEvent(handler.event, {handler: handler, error: error});
                  }
diff -crNB openidm-ui-ria-admin/org/forgerock/commons/ui/common/main/ServiceInvoker.js openidm-ui-ria-admin-apache/org/forgerock/commons/ui/common/main/ServiceInvoker.js
*** openidm-ui-ria-admin/org/forgerock/commons/ui/common/main/ServiceInvoker.js	2012-11-28 05:09:34.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/commons/ui/common/main/ServiceInvoker.js	2012-12-01 06:36:09.683630829 +0000
***************
*** 37,45 ****
      var obj = new AbstractConfigurationAware();
  
      obj.restCall = function(callParamsParam) {
!         var current = this, callParams, realSuccess, realError;
! 
!         callParamsParam.contentType = 'application/json; charset=utf-8';
          callParams = callParamsParam;
          obj.applyDefaultHeadersIfNecessary(callParams, obj.configuration.defaultHeaders);
  
--- 37,49 ----
      var obj = new AbstractConfigurationAware();
  
      obj.restCall = function(callParamsParam) {
!         var current = this, callParams, realSuccess, realError, nonJsonRequest = false;
!         
!         nonJsonRequest = (callParamsParam.hasOwnProperty('dataType') && callParamsParam.dataType !== "json");
!         
!         if (!nonJsonRequest) {
!             callParamsParam.contentType = 'application/json';
!         }
          callParams = callParamsParam;
          obj.applyDefaultHeadersIfNecessary(callParams, obj.configuration.defaultHeaders);
  
***************
*** 71,79 ****
                  realError(data);
              }
          };
!         callParams.dataType = "json";
!         callParams.contentType = "application/json";
! 
          $.ajax(callParams);	
      };
  
--- 75,84 ----
                  realError(data);
              }
          };
!         if (!nonJsonRequest) {
!             callParams.dataType = "json";
!             callParams.contentType = "application/json";
!         }
          $.ajax(callParams);	
      };
  
***************
*** 101,104 ****
      };
  
      return obj;
! });
\ No newline at end of file
--- 106,109 ----
      };
  
      return obj;
! });
diff -crNB openidm-ui-ria-admin/org/forgerock/commons/ui/user/delegates/UserDelegate.js openidm-ui-ria-admin-apache/org/forgerock/commons/ui/user/delegates/UserDelegate.js
*** openidm-ui-ria-admin/org/forgerock/commons/ui/user/delegates/UserDelegate.js	2012-11-28 05:09:50.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/commons/ui/user/delegates/UserDelegate.js	2012-12-01 06:14:11.406980798 +0000
***************
*** 31,38 ****
  	"org/forgerock/commons/ui/common/util/Constants",
      "org/forgerock/commons/ui/common/main/AbstractDelegate",
      "org/forgerock/commons/ui/common/main/Configuration",
!     "org/forgerock/commons/ui/common/main/EventManager"
! ], function(constants, AbstractDelegate, configuration, eventManager) {
  
      var obj = new AbstractDelegate(constants.host + "/openidm/managed/user");
  
--- 31,39 ----
  	"org/forgerock/commons/ui/common/util/Constants",
      "org/forgerock/commons/ui/common/main/AbstractDelegate",
      "org/forgerock/commons/ui/common/main/Configuration",
!     "org/forgerock/commons/ui/common/main/EventManager",
!     "org/forgerock/commons/ui/common/util/CookieHelper"
! ], function(constants, AbstractDelegate, configuration, eventManager, cookieHelper) {
  
      var obj = new AbstractDelegate(constants.host + "/openidm/managed/user");
  
***************
*** 58,84 ****
       * Starting session. Sending username and password to authenticate and returns user's id.
       */
      obj.login = function(uid, password, successCallback, errorCallback, errorsHandlers) {
!         var headers = {};
!         headers[constants.OPENIDM_HEADER_PARAM_USERNAME] = uid.toLowerCase();
!         headers[constants.OPENIDM_HEADER_PARAM_PASSWORD] = password;
!         headers[constants.OPENIDM_HEADER_PARAM_NO_SESION] = false;
          obj.serviceCall({
!             serviceUrl: constants.host + "/openidm/info/login",
              url: "",
!             headers: headers,
!             success: function (data) {
!                 if(!data.username) {
                      if(errorCallback) {
                          errorCallback();
                      }
!                 } else if(successCallback) {
!                     successCallback(data);
                  }
!             },
              error: errorCallback,
              errorsHandlers: errorsHandlers
          });
!         delete headers[constants.OPENIDM_HEADER_PARAM_PASSWORD];
      };
         
      /**
--- 59,86 ----
       * Starting session. Sending username and password to authenticate and returns user's id.
       */
      obj.login = function(uid, password, successCallback, errorCallback, errorsHandlers) {
! 
          obj.serviceCall({
!             serviceUrl: configuration.globalData.openamBaseURL + "/identity/authenticate?" + $.param({username: uid.toLowerCase(), password: password}),
              url: "",
!             dataType: "text",
!             type: "GET",
!             success: _.bind(function (data) {
! 		var token = "";
!                 if(!data.length || !data.match(/^token\.id=/)) {
                      if(errorCallback) {
                          errorCallback();
                      }
!                 } else {
! 			token = data.split("=")[1].replace("\n", "");
! 			cookieHelper.setCookie("iPlanetDirectoryPro", token, "", "", _.last(document.domain.split('.'), 2).join('.')); // the name of this cookie should probably be in the config as well 
! 			this.getProfile(successCallback, errorCallback, errorsHandlers);	
                  }
!             }, this),
              error: errorCallback,
              errorsHandlers: errorsHandlers
          });
! 	
      };
         
      /**
***************
*** 102,112 ****
       * Checks if logged in and returns users id
       */
      obj.getProfile = function(successCallback, errorCallback, errorsHandlers) {
          obj.serviceCall({
              serviceUrl: constants.host + "/openidm/info/login",
              url: "",
              success: function (data) {
!                 if(!data.username) {
                      if(errorCallback) {
                          errorCallback();
                      }
--- 104,122 ----
       * Checks if logged in and returns users id
       */
      obj.getProfile = function(successCallback, errorCallback, errorsHandlers) {
+ 
          obj.serviceCall({
+             serviceUrl: configuration.globalData.openamBaseURL + "/json/users/?_action=idFromSession",
+             data: "{}",
+             url: "",
+             type: "POST",
+             success: function (data) {
+ 
+        obj.serviceCall({
              serviceUrl: constants.host + "/openidm/info/login",
              url: "",
              success: function (data) {
!                 if(!data.username || data.username === "anonymous") {
                      if(errorCallback) {
                          errorCallback();
                      }
***************
*** 117,122 ****
--- 127,137 ----
              error: errorCallback,
              errorsHandlers: errorsHandlers
          });
+ 
+             },
+             error: errorCallback,
+             errorsHandlers: errorsHandlers
+         });
      };
      
      obj.readInternalEntity = function(id, successCallback, errorCallback) {
***************
*** 202,220 ****
      };
  
      obj.logout = function() {
-         var callParams = {
-                 url: "/",
-                 headers: {  },
-                 success: function () {
-                     console.debug("Successfully logged out");
-                 },
-                 error: function () {
-                     console.debug("Error during logging out");
-                 }
-         };
-         callParams.headers[constants.OPENIDM_HEADER_PARAM_LOGOUT] = true;
  
!         obj.serviceCall(callParams);
      };
  
      /**
--- 217,233 ----
      };
  
      obj.logout = function() {
  
!         obj.serviceCall({
!             serviceUrl: configuration.globalData.openamBaseURL + "/identity/logout?" + $.param({subjectid: cookieHelper.getCookie('iPlanetDirectoryPro')}),
!             url: "",
!             dataType: "text",
!             type: "GET",
!             success: function () { 
!                 cookieHelper.deleteCookie("iPlanetDirectoryPro", "", _.last(document.domain.split('.'), 2).join('.'));
!             }
! 	});
! 
      };
  
      /**
diff -crNB openidm-ui-ria-admin/org/forgerock/commons/ui/user/SiteConfigurator.js openidm-ui-ria-admin-apache/org/forgerock/commons/ui/user/SiteConfigurator.js
*** openidm-ui-ria-admin/org/forgerock/commons/ui/user/SiteConfigurator.js	2012-11-28 05:09:50.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/commons/ui/user/SiteConfigurator.js	2012-11-30 03:04:28.479111536 +0000
***************
*** 70,76 ****
         var router, changeSecurityDataDialog;
         
         router = require("org/forgerock/commons/ui/common/main/Router");
!        
         if(config.selfRegistration === true) {
             conf.globalData.selfRegistration = true;
         } else {               
--- 70,80 ----
         var router, changeSecurityDataDialog;
         
         router = require("org/forgerock/commons/ui/common/main/Router");
!       
!        if (config.hasOwnProperty('openamBaseURL')) {
!            conf.globalData.openamBaseURL = config.openamBaseURL;
!        } 
! 
         if(config.selfRegistration === true) {
             conf.globalData.selfRegistration = true;
         } else {               
***************
*** 131,134 ****
     };
     
     return obj;
! });
\ No newline at end of file
--- 135,138 ----
     };
     
     return obj;
! });
diff -crNB openidm-ui-ria-admin/org/forgerock/openam/ui/apps/AMDelegate.js openidm-ui-ria-admin-apache/org/forgerock/openam/ui/apps/AMDelegate.js
*** openidm-ui-ria-admin/org/forgerock/openam/ui/apps/AMDelegate.js	1970-01-01 00:00:00.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/openam/ui/apps/AMDelegate.js	2012-12-03 16:17:41.963963066 +0000
***************
*** 0 ****
--- 1,74 ----
+ /**
+  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+  *
+  * Copyright (c) 2011-2012 ForgeRock AS. All rights reserved.
+  *
+  * The contents of this file are subject to the terms
+  * of the Common Development and Distribution License
+  * (the License). You may not use this file except in
+  * compliance with the License.
+  *
+  * You can obtain a copy of the License at
+  * http://forgerock.org/license/CDDLv1.0.html
+  * See the License for the specific language governing
+  * permission and limitations under the License.
+  *
+  * When distributing Covered Code, include this CDDL
+  * Header Notice in each file and include the License file
+  * at http://forgerock.org/license/CDDLv1.0.html
+  * If applicable, add the following below the CDDL Header,
+  * with the fields enclosed by brackets [] replaced by
+  * your own identifying information:
+  * "Portions Copyrighted [year] [name of copyright owner]"
+  */
+ 
+ /*global $, define, _ */
+ 
+ /**
+  * @author jdabrowski
+  */
+ define("org/forgerock/openam/ui/apps/AMDelegate", [
+     "org/forgerock/commons/ui/common/util/Constants",
+     "org/forgerock/commons/ui/common/main/AbstractDelegate",
+     "org/forgerock/commons/ui/common/main/Configuration",
+     "org/forgerock/commons/ui/common/main/EventManager"
+ ], function(constants, AbstractDelegate, configuration, eventManager) {
+ 
+     var obj = {};
+ 
+     obj.getMyApplications = function (callback) {
+ 
+ 	if (!obj.d) {
+ 		obj.d = new AbstractDelegate(configuration.globalData.openamBaseURL);
+ 	}
+ 
+         obj.d.serviceCall({
+             url: "/json/dashboard/assigned",
+ 	    headers: {"Cache-Control": "no-cache"},
+             type: "GET",
+             success: callback
+         });
+         
+     };
+ 
+     obj.getAvailableApplications = function (callback) {
+ 
+         if (!obj.d) {
+                 obj.d = new AbstractDelegate(configuration.globalData.openamBaseURL);
+         }
+ 
+         obj.d.serviceCall({
+             url: "/json/dashboard/available",
+             headers: {"Cache-Control": "no-cache"},
+             type: "GET",
+             success: callback
+         });
+ 
+     };
+ 
+     
+     return obj;
+ });
+ 
+ 
+ 
diff -crNB openidm-ui-ria-admin/org/forgerock/openam/ui/apps/AppsDashboard.js openidm-ui-ria-admin-apache/org/forgerock/openam/ui/apps/AppsDashboard.js
*** openidm-ui-ria-admin/org/forgerock/openam/ui/apps/AppsDashboard.js	1970-01-01 00:00:00.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/openam/ui/apps/AppsDashboard.js	2012-11-30 22:16:09.948662131 +0000
***************
*** 0 ****
--- 1,58 ----
+ /**
+  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+  *
+  * Copyright (c) 2011-2012 ForgeRock AS. All rights reserved.
+  *
+  * The contents of this file are subject to the terms
+  * of the Common Development and Distribution License
+  * (the License). You may not use this file except in
+  * compliance with the License.
+  *
+  * You can obtain a copy of the License at
+  * http://forgerock.org/license/CDDLv1.0.html
+  * See the License for the specific language governing
+  * permission and limitations under the License.
+  *
+  * When distributing Covered Code, include this CDDL
+  * Header Notice in each file and include the License file
+  * at http://forgerock.org/license/CDDLv1.0.html
+  * If applicable, add the following below the CDDL Header,
+  * with the fields enclosed by brackets [] replaced by
+  * your own identifying information:
+  * "Portions Copyrighted [year] [name of copyright owner]"
+  */
+ 
+ /*global define*/
+ 
+ define("org/forgerock/openam/ui/apps/AppsDashboard", [
+     "org/forgerock/commons/ui/common/main/AbstractView",
+     "org/forgerock/openam/ui/apps/AMDelegate",
+     "org/forgerock/commons/ui/common/main/EventManager",
+     "org/forgerock/commons/ui/common/util/Constants",
+     "org/forgerock/commons/ui/common/main/Configuration"
+ ], function(AbstractView, AMDelegate, eventManager, constants, conf) { 
+     
+     var Dashboard = AbstractView.extend({
+         template: "templates/apps/DashboardTemplate.html",
+         element: "#appsList",
+         render: function() {
+             
+             AMDelegate.getMyApplications(_.bind(function (apps) {
+ 		var sortedApps = _.map(_.sortBy(_.keys(apps), function (key){ return key; }), function (key) { 
+ 	
+ 			var app = {}; 
+ 			app.id = key; 
+ 			_.each(apps[key], function (v,k) { app[k] = v[0]; });
+ 			return app; 
+ 
+ 		});
+ 		this.data = {};
+                 this.data.apps = sortedApps;
+                 this.parentRender();
+             }, this));
+             
+         }
+     });
+     
+     return Dashboard;
+ });
diff -crNB openidm-ui-ria-admin/org/forgerock/openam/ui/apps/main.js openidm-ui-ria-admin-apache/org/forgerock/openam/ui/apps/main.js
*** openidm-ui-ria-admin/org/forgerock/openam/ui/apps/main.js	1970-01-01 00:00:00.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/openam/ui/apps/main.js	2012-11-30 03:34:09.232344525 +0000
***************
*** 0 ****
--- 1,29 ----
+ /**
+  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+  *
+  * Copyright (c) 2011-2012 ForgeRock AS. All rights reserved.
+  *
+  * The contents of this file are subject to the terms
+  * of the Common Development and Distribution License
+  * (the License). You may not use this file except in
+  * compliance with the License.
+  *
+  * You can obtain a copy of the License at
+  * http://forgerock.org/license/CDDLv1.0.html
+  * See the License for the specific language governing
+  * permission and limitations under the License.
+  *
+  * When distributing Covered Code, include this CDDL
+  * Header Notice in each file and include the License file
+  * at http://forgerock.org/license/CDDLv1.0.html
+  * If applicable, add the following below the CDDL Header,
+  * with the fields enclosed by brackets [] replaced by
+  * your own identifying information:
+  * "Portions Copyrighted [year] [name of copyright owner]"
+  */
+ 
+ /*global define*/
+ 
+ define("org/forgerock/openam/ui/apps/main", [
+     "./AppsDashboard"
+ ]);
\ No newline at end of file
diff -crNB openidm-ui-ria-admin/org/forgerock/openidm/ui/admin/workflow/tasks/customview/ApplicationAcceptanceTask.js openidm-ui-ria-admin-apache/org/forgerock/openidm/ui/admin/workflow/tasks/customview/ApplicationAcceptanceTask.js
*** openidm-ui-ria-admin/org/forgerock/openidm/ui/admin/workflow/tasks/customview/ApplicationAcceptanceTask.js	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/openidm/ui/admin/workflow/tasks/customview/ApplicationAcceptanceTask.js	2012-12-03 00:32:58.793632100 +0000
***************
*** 41,53 ****
              js2form(document.getElementById(this.$el.attr("id")), this.task);
              this.$el.find("input[name=taskName]").val(this.task.name);
              this.$el.find("input[name=createTime]").val(DateUtil.formatDate(this.task.createTime));
!             
              if(this.$el.find("input[name=assignee]").val() === "null") {
                  this.$el.find("input[name=assignee]").val("");
              }
!             
!             this.$el.find("input[name=userData]").val(this.task.variables.user.givenName + " " + this.task.variables.user.familyName);
!             this.$el.find("input[name=requestedApplicationName]").val(this.task.variables.application.name);            
          }
      }); 
      
--- 41,53 ----
              js2form(document.getElementById(this.$el.attr("id")), this.task);
              this.$el.find("input[name=taskName]").val(this.task.name);
              this.$el.find("input[name=createTime]").val(DateUtil.formatDate(this.task.createTime));
!            /* 
              if(this.$el.find("input[name=assignee]").val() === "null") {
                  this.$el.find("input[name=assignee]").val("");
              }
!            */
!             this.$el.find("input[name=userData]").val(this.task.startUserDisplayable);
!             this.$el.find("input[name=requestedApplicationName]").val(this.task.variables.lnkId);            
          }
      }); 
      
diff -crNB openidm-ui-ria-admin/org/forgerock/openidm/ui/admin/workflow/tasks/TasksDashboard.js openidm-ui-ria-admin-apache/org/forgerock/openidm/ui/admin/workflow/tasks/TasksDashboard.js
*** openidm-ui-ria-admin/org/forgerock/openidm/ui/admin/workflow/tasks/TasksDashboard.js	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/org/forgerock/openidm/ui/admin/workflow/tasks/TasksDashboard.js	2012-11-30 03:36:23.432055881 +0000
***************
*** 37,45 ****
      "org/forgerock/commons/ui/common/main/Configuration",
      "org/forgerock/openidm/ui/admin/notifications/NotificationDelegate",
      "org/forgerock/openidm/ui/admin/workflow/tasks/TaskDetailsView",
!     "org/forgerock/openidm/ui/admin/workflow/processes/StartProcessDashboardView"
  ], function(AbstractView, workflowManager, eventManager, constants, TasksMenuView, 
!     NotificationsView, conf, notificationDelegate, taskDetailsView, startProcessView) {
      
      var TasksDashboard = AbstractView.extend({
          template: "templates/admin/workflow/tasks/TasksDashboardTemplate.html",
--- 37,46 ----
      "org/forgerock/commons/ui/common/main/Configuration",
      "org/forgerock/openidm/ui/admin/notifications/NotificationDelegate",
      "org/forgerock/openidm/ui/admin/workflow/tasks/TaskDetailsView",
!     "org/forgerock/openidm/ui/admin/workflow/processes/StartProcessDashboardView",
!     "org/forgerock/openam/ui/apps/AppsDashboard"
  ], function(AbstractView, workflowManager, eventManager, constants, TasksMenuView, 
!     NotificationsView, conf, notificationDelegate, taskDetailsView, startProcessView, AppsDashboard) {
      
      var TasksDashboard = AbstractView.extend({
          template: "templates/admin/workflow/tasks/TasksDashboardTemplate.html",
***************
*** 54,65 ****
              
              this.parentRender(function() {
                  var notificationsView;
!                 
                  this.candidateTasks.render("all", $("#candidateTasks"));                
                  this.myTasks.render("assigned", $("#myTasks"));
                  startProcessView.render();
                  
-                 if (params.mode === "user") {
                      //notifications
                      notificationDelegate.getNotificationsForUser(function(notifications) {
                          
--- 55,68 ----
              
              this.parentRender(function() {
                  var notificationsView;
! 
!                 this.appsDashboard = new AppsDashboard();
! 
!                 this.appsDashboard.render();
                  this.candidateTasks.render("all", $("#candidateTasks"));                
                  this.myTasks.render("assigned", $("#myTasks"));
                  startProcessView.render();
                  
                      //notifications
                      notificationDelegate.getNotificationsForUser(function(notifications) {
                          
***************
*** 76,82 ****
                          notificationsView = new NotificationsView();
                          notificationsView.render({el: $("#notifications"), items: notifications});
                      });
!                 }
                  if (args && args[0] && args[0] !== '') {
                      this.showDetails({id: args[0]});
                  }
--- 79,85 ----
                          notificationsView = new NotificationsView();
                          notificationsView.render({el: $("#notifications"), items: notifications});
                      });
! 
                  if (args && args[0] && args[0] !== '') {
                      this.showDetails({id: args[0]});
                  }
diff -crNB openidm-ui-ria-admin/templates/admin/workflow/tasks/customview/ApplicationAcceptanceTemplate.html openidm-ui-ria-admin-apache/templates/admin/workflow/tasks/customview/ApplicationAcceptanceTemplate.html
*** openidm-ui-ria-admin/templates/admin/workflow/tasks/customview/ApplicationAcceptanceTemplate.html	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/templates/admin/workflow/tasks/customview/ApplicationAcceptanceTemplate.html	2012-12-03 00:35:40.950198819 +0000
***************
*** 1,22 ****
! <div class="field">
!     <label class="light">{{t "common.task.taskName"}}</label> 
!     <input type="text" name="taskName" /> 
! </div>
  <div class="field">
      <label class="light">{{t "common.application.applicationName"}}</label> 
!     <input type="text" name="requestedApplicationName" /> 
  </div>
  <div class="field">
      <label class="light">{{t "common.task.assignee"}}</label> 
!     <input type="text" name="assignee" /> 
  </div>
  <div class="field">
      <label class="light">{{t "common.user.user"}}</label> 
!     <input type="text" name="userData" /> 
  </div>
  <div class="field">
      <label class="light">{{t "common.form.createdAt"}}</label> 
!     <input type="text" name="createTime" /> 
  </div>
  <div class="field">
      <label class="light">{{t "common.form.decision"}}</label> 
--- 1,23 ----
! 
! <!--
  <div class="field">
      <label class="light">{{t "common.application.applicationName"}}</label> 
!     <input type="text" name="requestedApplicationName" readonly /> 
  </div>
+ 
  <div class="field">
      <label class="light">{{t "common.task.assignee"}}</label> 
!     <input type="text" name="assignee" readonly /> 
  </div>
+ -->
+ 
  <div class="field">
      <label class="light">{{t "common.user.user"}}</label> 
!     <input type="text" name="userData" readonly /> 
  </div>
  <div class="field">
      <label class="light">{{t "common.form.createdAt"}}</label> 
!     <input type="text" name="createTime" readonly /> 
  </div>
  <div class="field">
      <label class="light">{{t "common.form.decision"}}</label> 
diff -crNB openidm-ui-ria-admin/templates/admin/workflow/tasks/TasksDashboardTemplate.html openidm-ui-ria-admin-apache/templates/admin/workflow/tasks/TasksDashboardTemplate.html
*** openidm-ui-ria-admin/templates/admin/workflow/tasks/TasksDashboardTemplate.html	2012-11-28 17:22:00.000000000 +0000
--- openidm-ui-ria-admin-apache/templates/admin/workflow/tasks/TasksDashboardTemplate.html	2012-11-30 03:37:44.542299019 +0000
***************
*** 1,6 ****
--- 1,9 ----
  <div class="content" id="task-dashboard">
  	<div class="column-layout">
  	    <div class="column5">
+             <h2 class="header">{{t "openam.apps.header"}}</h2>
+             <div id="appsList" style="margin-bottom: 20px"></div>
+ 
  	        <h2 class="header">{{t "common.task.myTasks"}}</h2>
  	        <div id="myTasks" class="tasks-table"></div>
  	        
diff -crNB openidm-ui-ria-admin/templates/apps/DashboardTemplate.html openidm-ui-ria-admin-apache/templates/apps/DashboardTemplate.html
*** openidm-ui-ria-admin/templates/apps/DashboardTemplate.html	1970-01-01 00:00:00.000000000 +0000
--- openidm-ui-ria-admin-apache/templates/apps/DashboardTemplate.html	2012-11-30 22:30:23.152415023 +0000
***************
*** 0 ****
--- 1,5 ----
+ <ul class="appsDashboard">
+ {{#each apps}}
+     <li><a href="{{dashboardLogin}}"><img src="{{dashboardIcon}}">{{dashboardName}}</a></li>
+ {{/each}}
+ </ul>
